coap-core/src/main/java/com/mbed/coap/transport/javassl/SocketClientTransport.java

```diff
@@ -81,6 +91,7 @@ protected void connect(CoapReceiver coapReceiver) throws IOException {
     }
 
     private boolean loopReading(CoapReceiver coapReceiver) {
+        InetSocketAddress destination = destinationSupplier.get();
```

Comment:
should this be read only after `waitBeforeReconnection` ?

Reply:
fixed.